src/main.ts

```diff
@@ -0,0 +1,251 @@
+ * primary entry point. It is documented inline.
+ */
+export async function run(): Promise<void> {
+  const restoreEnv = stubEnv({
+    CLOUDSDK_METRICS_ENVIRONMENT: 'github-actions-create-cloud-deploy-release',
+    CLOUDSDK_METRICS_ENVIRONMENT_VERSION: appVersion,
+  });
+
+  // Warn if pinned to HEAD
+  if (isPinnedToHead()) {
+    logWarning(pinnedToHeadWarning('v1'));
+  }
+  try {
+    // Core inputs (required)
+    const name = getInput('name');
+    const deliveryPipeline = getInput('delivery_pipeline');
+    const region = getInput('region');
+    const source = getInput('source');
+    const buildArtifacts = getInput('build_artifacts');
+    const images = parseKVString(getInput('images'));
+    const disableInitialRollout = getBooleanInput('disable_initial_rollout');
+    const sourceStagingDir = getInput('gcs_source_staging_dir');
+    const skaffoldFile = getInput('skaffold_file');
+    const annotations = parseKVString(getInput('annotations'));
+    const labels = parseKVString(getInput('labels'));
+    const description = getInput('description');
+    const flags = getInput('flags');
+    const gcloudComponent = presence(getInput('gcloud_component'));
+    const gcloudVersion = await computeGcloudVersion(getInput('gcloud_version'));
+
+    let cmd;
+
+    // Throw errors if inputs aren't valid
+    if (!name) {
+      throw new Error('No release name set.');
+    }
+    if (!deliveryPipeline) {
+      throw new Error('No delivery pipeline set.');
+    }
+    if (!region) {
+      throw new Error('No region set.');
+    }
+    if (!buildArtifacts && !images) {
+      throw new Error('One of `build_artifacts` and `images` inputs must be supplied.');
+    }
+    if (buildArtifacts && images && Object.keys(images).length > 0) {
+      throw new Error('Both `build_artifacts` and `images` inputs set - please select only one.');
+    }
+
+    // Validate gcloud component input
+    if (gcloudComponent && gcloudComponent !== 'alpha' && gcloudComponent !== 'beta') {
+      throw new Error(`invalid input received for gcloud_component: ${gcloudComponent}`);
+    }
+
+    // Build base command from required inputs
+    cmd = [
+      'deploy',
+      'releases',
+      'create',
+      name,
+      '--delivery-pipeline',
+      deliveryPipeline,
+      '--region',
+      region,
+    ];
+
+    if (buildArtifacts) {
+      cmd.push('--build-artifacts', buildArtifacts);
+    }
+    if (images && Object.keys(images).length > 0) {
+      cmd.push('--images', kvToString(images));
+    }
+    if (source) {
+      cmd.push('--source', source);
+    }
+    if (disableInitialRollout) {
+      cmd.push('--disable-initial-rollout');
+    }
+    if (sourceStagingDir) {
+      cmd.push('--gcs-source-staging-dir', sourceStagingDir);
+    }
+    if (skaffoldFile) {
+      cmd.push('--skaffold-file', skaffoldFile);
+    }
+    if (annotations) {
+      const allAnnotations = Object.assign({}, getDefaultAnnotations(context), annotations);
```

Comment:
Don't we always want to assign the default annotations, even if no user-supplied annotations exist?

Reply:
You're right -- I will also add a test to check this.